### PR TITLE
Improve tutorial section on differential expression testing between clonotypes

### PR DESCRIPTION
**Pull request by szabogtamas**

### CODE REVIEW

File: docs/tutorials/tutorial_3k_tcr.md [hunk 1]
```diff
@@ -604,35 +604,54 @@ ir.pl.clonotype_imbalance(
...
-    case_label="Tumor",
-    control_label="Blood",
-    additional_hue="cluster",
+    groupby="cluster",
+    case_label="CD8_Teff",
+    control_label="CD8_Trm",
     inplace = False,
 )
-top_differential_clonotypes = stat.groupby("clonotype").agg({"logpValue": "max"}).reset_index().sort_values(by="logpValue", ascending=False).head(n=15).clonotype.tolist()
+top_differential_clonotypes = stat.groupby("clonotype").agg({"logpValue": "max"}).reset_index().sort_values(by="logpValue", ascending=False).head(n=5).clonotype.tolist()
```
Comment (grst) [inline docs/tutorials/tutorial_3k_tcr.md:618]: as the table is already ordered, this can be simpliefied to 
```
stat["clonotype"][:5].values
```

Comment (szabogtamas) [reply docs/tutorials/tutorial_3k_tcr.md:618]: True, this was only needed for the previous case, where we had two grouping factors.

File: docs/tutorials/tutorial_3k_tcr.md [hunk 2]
```diff
@@ -604,35 +604,54 @@ ir.pl.clonotype_imbalance(
...
 ```
 
-To check how specific these top clonotypes are for individual cell types, a UMAP is straighforward.
+Showing top clonotypes on a UMAP clearly shows that clonotype 163 is featured by CD8+ tissue-resident memory T cells, while clonotype 277 by CD8+ effector T-cells.
 
 ```python
 sc.pl.umap(adata, color="cluster")
```
Comment (grst) [inline docs/tutorials/tutorial_3k_tcr.md:624]: Cosmetics: 
```python
fig, (ax1, ax2) = plt.subplots(1, 2, figsize=(12, 4), gridspec_kw={"wspace": 0.6})
sc.pl.umap(adata, color="cluster", ax=ax1, show=False)
sc.pl.umap(
    adata,
    color="clonotype",
    groups=top_differential_clonotypes,
    ax=ax2,
    # increase size of highlighted dots
    size=[
        80 if c in top_differential_clonotypes else 30 for c in adata.obs["clonotype"]
    ],
)
```

image

File: docs/tutorials/tutorial_3k_tcr.md [hunk 3]
```diff
@@ -604,35 +604,51 @@ ir.pl.clonotype_imbalance(
 ## Integrating gene expression
 ### Clonotype imbalance among cell clusters
 
-Leveraging the opportunity offered by close integeration with scanpy, transcriptomics-based data can be utilized directly. As an example, using cell type annotation inferred from gene expression clusters, clonotypes belonging to CD4+ and CD8+ regional memory T-cells can be compared.
+Leveraging the opportunity offered by close integeration with scanpy, transcriptomics-based data can be utilized directly. As an example, using cell type annotation inferred from gene expression clusters, clonotypes belonging to CD8+ effector T-cells and CD8+ tissue-resident memory T cells, for example, can be compared.
```
Comment (grst) [inline docs/tutorials/tutorial_3k_tcr.md:607]:
```suggestion
Leveraging the opportunity offered by close integeration with scanpy, transcriptomics-based data can be utilized directly. Using cell type annotation inferred from gene expression clusters, for example, clonotypes belonging to CD8+ effector T-cells and CD8+ tissue-resident memory T cells, can be compared.
```

**Comment by grst** [comment]
Ready to merge?

**Comment by szabogtamas** [comment]
Yes!

Gregor Sturm <notifications@github.com> ezt írta (időpont: 2020. jún. 5.,
P, 10:53):

**Pull request merged**